e2e_tests/tests/per_provider/normal_tests/asym_encryption.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use e2e_tests::TestClient;
+use parsec_client::core::interface::requests::{ProviderID, ResponseStatus};
+use rsa::{RSAPublicKey, PaddingScheme, PublicKey};
+use rand::rngs::OsRng;
+
+
+const PLAINTEXT_MESSAGE: [u8; 32] = [
+    0x69, 0x3E, 0xDB, 0x1B, 0x22, 0x79, 0x03, 0xF4, 0xC0, 0xBF, 0xD6, 0x91, 0x76, 0x37, 0x84, 0xA2,
+    0x94, 0x8E, 0x92, 0x50, 0x35, 0xC2, 0x8C, 0x5C, 0x3C, 0xCA, 0xFE, 0x18, 0xE8, 0x81, 0x37, 0x78,
+];
+
+#[test]
+fn simple_asym_encrypt_rsa_pkcs() {
+    let key_name = String::from("asym_encrypt_and_decrypt_rsa_pkcs");
+    let mut client = TestClient::new();
+
+    if client.provider() != Some(ProviderID::MbedCrypto) {
+        // Not supported by current provider
+        return;
+    }
```

Comment:
Is there a neater way of doing this? 
Would be nice if the tests could check which operations the current provider supports and only test those.

Reply:
You can always put an `#[cfg(feature = "mbed-crypto-provider")]` on the test

Reply:
Ah, this is in the e2e crate... Hmmm

Reply:
I don't think the `if` is 100% correct anyway, because you could have multiple providers set up and the client chooses a different one, doesn't mean the Mbed Crypto one isn't available. Maybe we should change the `set_provider` method to check if the provider actually exists, and returns an error if not. Then we can check on that.

Reply:
> client chooses a different one

Are the tests not done in isolation? As in, there will only ever be one provider at once during testing?

> Maybe we should change the set_provider method (on the test client) to check if the provider actually exists, and returns an error if not. Then we can check on that.

So have `set_provider` check `basic_client.list_providers` to see if that provider exists (in this case, `MbedCrypto`) and if it does, it returns `Ok` and we can run the tests?

Reply:
> Are the tests not done in isolation? As in, there will only ever be one provider at once during testing?

Yeah, that's true, this is in the per_provider test thing - forget that.

I think we can do it in another way, actually, makes it more scalable - call `list_opcodes` and check if the operation you want is supported. If it isn't, `return`. It's not ideal because it will still appear as a test even when it's not supported, but there's no need to do anything to enable the test when you implement it in some new provider.

Reply:
> I think we can do it in another way, actually, makes it more scalable - call list_opcodes and check if the operation you want is supported. If it isn't, return. It's not ideal because it will still appear as a test even when it's not supported, but there's no need to do anything to enable the test when you implement it in some new provider.

That sounds more like what I was thinking. The other two solutions don't deal with the fact that the tests are still run either. Could do with a way of forcing text output even if `--show-output` isn't specified, so we can at least say 'Hey, this provider doesn't actually support this operation'. I can have a poke tomorrow.